pkg/proxy/iptables/proxier.go

```diff
@@ -1387,23 +1394,52 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 	}
 
-	// Finally, tail-call to the nodeports chain.  This needs to be after all
+	// Finally, tail-call to the nodePorts chain.  This needs to be after all
 	// other service portal rules.
 	for address := range nodeAddresses {
 		if utilproxy.IsZeroCIDR(address) {
+			destinations := []string{"-m", "addrtype", "--dst-type", "LOCAL"}
```

Comment:
you don't need to add new rules, just set `route_localnet` to 0 on line 236, that options is that makes the nodePort available on localhost, disabling it they won't work

Reply:
@aojea we still need generate iptables rules that prevent localhost nodeport connections when flag is false. please check this discussion https://github.com/kubernetes/kubernetes/pull/108250#discussion_r893572847

Reply:
oh, sorry , missed that discussion, thanks for pointing it out